frontend/src/components/Per-ImplementationPage/PerImplementationPage.tsx

```diff
@@ -0,0 +1,108 @@
+import { Card } from "react-bootstrap";
+import { useLoaderData, Link } from "react-router-dom";
+import { Table, Container } from "react-bootstrap";
+import { ReportData, ImplementationMetadata } from "../../data/parseReportData";
+// @ts-ignore
+import LoadingAnimation from "../../components/LoadingAnimation";
+import DialectCompliance from "./DialectCompliance";
+
+export const PerImplementationPage = () => {
+  const loaderData = useLoaderData() as ReportData;
+
+  const pathSegments = window.location.href.split("/");
+  let implementationName = pathSegments[pathSegments.length - 1];
+
+  let implementationDetail: ImplementationMetadata = {};
+  let allImplementations = {};
+  Object.values(loaderData).map((value) => {
+    allImplementations = {
+      ...allImplementations,
+      ...value.runInfo.implementations,
+    };
+  });
+  Object.keys(allImplementations).forEach((key) => {
+    if (key.includes(implementationName)) {
+      implementationDetail = allImplementations[key];
+    }
+  });
+  return implementationDetail ? (
```

Comment:
This 80+ lines ternary operator is not really readable :P

Reply:
Moved those lines into its own component for improved readability! 😄
```js
  return implementationDetail ? (
    <ReportComponent />
  ) : (
    <LoadingAnimation />
  );
```